Lib/test/test_socket.py

```diff
@@ -7348,21 +7351,115 @@ def close_fds(fds):
+            self.assertEqual(flags & socket.MSG_CTRUNC, 0)
+        self.assertEqual(len(fds), 1)
+        self.assertEqual(addr, sock1_addr)
+
+        self._test_pipe(fds[0], wfd, MSG)
 
-        for index, rfd in enumerate(fds2):
-            data = os.read(rfd, 100)
-            self.assertEqual(data,  str(index).encode())
+    @requireAttrs(socket, "MSG_PEEK")
+    @unittest.skipUnless(sys.platform in ("linux", "android"), "works on Linux")
+    def test_recv_fds_peek(self):
+        rfd, wfd = os.pipe()
+        self.addCleanup(os.close, rfd)
+        self.addCleanup(os.close, wfd)
+
+        sock1, sock2 = socket.socketpair(socket.AF_UNIX, socket.SOCK_DGRAM)
+        with sock1, sock2:
+            socket.send_fds(sock1, [MSG], [rfd])
+            sock2.setblocking(False)
+
+            # peek message on sock2
+            peek_len = len(MSG) // 2
+            msg, fds, flags, addr = socket.recv_fds(sock2, peek_len, 1,
+                                                    flags=socket.MSG_PEEK)
+            self._cleanup_fds(fds)
+
+            self.assertEqual(len(msg), peek_len)
+            self.assertEqual(msg, MSG[:peek_len])
+            self.assertEqual(flags & socket.MSG_TRUNC, socket.MSG_TRUNC)
+            if hasattr(socket, "MSG_CTRUNC"):
+                self.assertEqual(flags & socket.MSG_CTRUNC, 0)
+            self.assertEqual(len(fds), 1)
+            self._test_pipe(fds[0], wfd, MSG)
+
+            # will raise BlockingIOError if MSG_PEEK didn't work
+            msg, fds, flags, addr = socket.recv_fds(sock2, len(MSG), 1)
+            self._cleanup_fds(fds)
+
+            self.assertEqual(msg, MSG)
+            if hasattr(socket, "MSG_CTRUNC"):
+                self.assertEqual(flags & socket.MSG_CTRUNC, 0)
+            self.assertEqual(len(fds), 1)
+            self._test_pipe(fds[0], wfd, MSG)
+
+    @requireAttrs(socket, "MSG_DONTWAIT")
+    @unittest.skipIf(sys.platform in ("darwin",),
+                     "MSG_DONTWAIT not known to work on this platform when "
+                     "sending")
+    def test_send_fds_dontwait(self):
+        rfd, wfd = os.pipe()
+        self.addCleanup(os.close, rfd)
+        self.addCleanup(os.close, wfd)
+
+        # use SOCK_STREAM instead of SOCK_DGRAM to support *BSD platforms
+        # ref: https://docs.python.org/3/library/asyncio-protocol.html#datagram-protocols
+        sock1, sock2 = socket.socketpair(socket.AF_UNIX, socket.SOCK_STREAM)
+        with sock1, sock2:
+            sock1.setblocking(True)
+            with self.assertRaises(BlockingIOError):
+                for _ in range(64 * 1024):
+                    socket.send_fds(sock1, [MSG], [rfd], socket.MSG_DONTWAIT)
+
+            if sys.platform.startswith("freebsd"):
+                # FreeBSD requires at least CMSG_LEN(2 * sizeof(int)), otherwise
+                # the cmsg will be truncated
+                recv_fds_len = 2
+            else:
+                recv_fds_len = 1
```

Comment:
Maybe prefix the variable by `max_` (ditto for the other place) as it's the `max_fds` parameter.